Userland/Libraries/LibCore/Stream.h

```diff
@@ -724,37 +708,72 @@ class BufferedHelper {
         return m_buffer.read(buffer);
     }
 
-    // Returns whether a line can be read, populating the buffer in the process.
-    ErrorOr<bool> can_read_line()
+    struct OffsetAndSize {
+        size_t offset {};
+        size_t match_size {};
+    };
```

Comment:
Also, a very minor nitpick, but this feels quite verbose. How about the following?

```suggestion
    struct Match {
        size_t offset {};
        size_t size {};
    };
```